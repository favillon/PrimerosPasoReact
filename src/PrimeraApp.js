// Funtional components

import {Fragment} from "react";
import PropTypes from "prop-types";

const PrimeraApp = ( { saludo='Hola Mundo', subtitulo} ) => {
    
    return (
        <Fragment>
            <h1>{ saludo }</h1>
            <p>{ subtitulo }</p>
        </Fragment>
    );
}

PrimeraApp.propType = {
    saludo : PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
    subtitulo : 'Soy un subtitulo, Mi Primera Aplicacion'
}

export default PrimeraApp;