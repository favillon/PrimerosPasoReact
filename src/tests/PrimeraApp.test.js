import React from "react";
//import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from "../PrimeraApp";


describe('Pruebas <PrimeraApp /> ', ()=>{

    // jests
    // test('Debe mostrar Hola Soy Goku', () =>{
    //     // Inicializacion 
    //     const saludo = 'Hola Soy Goku';

    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);

    //     expect( getByText( saludo ) ).toBeInTheDocument();
    
    // })

    test('Debe mostrar Hola Soy Goku', () =>{
        // Inicializacion 
        const saludo='Hola Soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();
    
    })
    test('Debe mostrar un subtitulo', () =>{
        // Inicializacion 
        const saludo='Hola Soy Goku';
        const sub='Sayaying';
        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo={sub} />
        );

        const textoParrafo = wrapper.find('p').text();
        
        expect(textoParrafo).toBe(sub);
    
    })
})
