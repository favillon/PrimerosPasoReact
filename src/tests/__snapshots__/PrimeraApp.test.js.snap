// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <PrimeraApp />  Debe mostrar Hola Soy Goku 1`] = `
<Fragment>
  <h1>
    Hola Soy Goku
  </h1>
  <p>
    Soy un subtitulo, Mi Primera Aplicacion
  </p>
</Fragment>
`;
